src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor_Minimal.cs

```diff
@@ -159,6 +159,20 @@ private void MinimallyQualify(INamedTypeSymbol symbol)
             AddNameAndTypeArgumentsOrParameters(symbol);
         }
 
+        protected bool InObjectInitializatioContext(INamedTypeSymbol symbol)
```

Comment:
I'm absolutely fine with this behavior changing.  "yield yield" is totally ok.  This scenario is completely irrelevant :)